### DISCUSSION TITLE
Huge inconsistencies in patientweight in inputevents

### DISCUSSION THREAD

**Discussion by mpierrau** [MIT-LCP/mimic-code#1535] [MIMIC-IV]
Hi!

I have been working on MIMIC-IV for some time and recently discovered that the column `patientweight` has some extreme inconsistencies for some `stay_id` which I fail to find a reasonable explanation. 

There are multiple rows with `starttime` within minutes but with different `patientweight`. I understand this can happen, but the problem is that some differences are extremely big -- like tens of kilos difference. 

I've been parsing MIMIC-IV using Python so I don't have a nice reproducible SQL code for you, but you can have a look at for example `stay_id`'s:
39615949 (diff 170 kg between starttime 2145-10-06 20:52:00 and starttime 2145-10-06 22:31:00)
35032951 (diff 23 kg)
38680231 (diff 25 kg)
33230840 (diff 107 kg)
...

Can you provide a reason for why this is?

Is daily weight (itemid 224639 in chartevents) or admission weight (itemid 226512 in chartevents) a better option?

Kindly,
Magnus

**Comment by yongfanbeta** [comment]
When I check this problem through the following steps 

```sql
select *
  from chartevents
 where stay_id = '39615949'
   and itemid = '224639'
 order by charttime
```

subject_id | hadm_id | stay_id | caregiver_id | charttime | storetime | itemid | value | valuenum | valueuom | warning
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
10092175 | 27558426 | 39615949 | 67543 | 2145-10-06 03:00:00 | 2145-10-06 02:31:00 | 224639 | 234 | 234.0 | kg | 0

(only show some result)
Question1: storetime is less than charttime

```sql
select *
from inputevents
where stay_id = '39615949'
order by starttime
```

subject_id | hadm_id | stay_id | caregiver_id | starttime | endtime | storetime | itemid | amount | amountuom | rate | rateuom | orderid | linkorderid | ordercategoryname | secondaryordercategoryname | ordercomponenttypedescription | ordercategorydescription | patientweight | totalamount | totalamountuom | isopenbag | continueinnextdept | statusdescription | originalamount | originalrate
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
10092175 | 27558426 | 39615949 | 67543 | 2145-10-06 03:00:00 | 2145-10-06 03:01:00 | 2145-10-06 04:04:00 | 225152 | 1700.000000 | units | NaN | None | 2036700 | 2036700 | 05-Med Bolus | None | Main order parameter | Drug Push | 400.0 | NaN | None | 0 | 0 | FinishedRunning | 1700.000000 | 1700.000000

Question2: patientweight in `inputevents` is inconsistent with weight in `chartevents` nearly the same time

Question3: as @mpierrau said, patientweight results from inputevents in stay_id = '39615949' has huge fluctuation. Is it normal or error?
